### PR TITLE
UI revamp

### PR DESCRIPTION
**Pull request by kshitijk4poor**
# 🎨 UI revamp

## 🚀 Changes

- 🖌️ Integrated the Zephyr theme from Bootswatch ([bootswatch.com/zephyr/](https://bootswatch.com/zephyr/))
- 🔧 Fixed navbar issues, now it correctly shows the current active states for the navbar items
- ☀️ Supports both Light and Dark themes

## 🌟 Benefits

- 💅 Fresh, modern look with the "breezy and beautiful" Zephyr theme
- 📱 Enhanced user experience with responsive navbar
- 🎭 Improved overall UI consistency across the application

## 🧪 Testing

- Verified theme application across all pages
- Tested navbar functionality on various screen sizes
- Ensured no conflicts with existing custom styles

## ☑️ TODO
- [x] Implement theme toggling
- [ ] add dark mode for `queue.jinja2` (needs to be done on karton repo)

### CODE REVIEW

**Comment by kazet** [comment]
screnshot: censor the IP ;)

File: templates/components/base.jinja2 [hunk 1]
```diff
@@ -19,9 +18,10 @@
             {% endblock %}
         </main>
         {% endblock %}
-
+        <script src="/static/theme.js"></script>
+        <script src="https://kit.fontawesome.com/822c3cc402.js" crossorigin="anonymous"></script>
```
Comment (kazet) [inline templates/components/base.jinja2:22]: what is this? can we integrate this locally without adding one more js dependency?

File: templates/components/navbar.jinja2 [hunk 2]
```diff
@@ -1,35 +1,43 @@
...
-                        <a class="nav-link" href="/add">Add targets</a>
+                        <a class="nav-link {{ 'active' if request.path == '/add' }}" href="/add">Add targets</a>
                     </li>
                     <li class="nav-item">
-                        <a class="nav-link" href="/">View targets</a>
+                        <a class="nav-link {{ 'active' if request.path == '/' }}" href="/">View targets</a>
                     </li>
                     <li class="nav-item">
-                        <a class="nav-link" href="/results?task_filter=interesting">View raw task results</a>
+                        <a class="nav-link {{ 'active' if request.path == '/results' }}" href="/results?task_filter=interesting">View raw task results</a>
```
Comment (kazet) [inline templates/components/navbar.jinja2:17]: would it work if the path is e.g. /results?task_filter=interesting ?

File: templates/components/navbar.jinja2 [hunk 3]
```diff
@@ -1,35 +1,43 @@
...
+                        <a class="nav-link {{ 'active' if request.path == '/restart-crashed-tasks' }}" href="/restart-crashed-tasks">Restart crashed tasks</a>
                     </li>
                     <li class="nav-item">
-                        <a class="nav-link" href="/docs">API</a>
+                        <a class="nav-link {{ 'active' if request.path == '/docs' }}" href="/docs">API</a>
+                    </li>
+                    <li class="nav-item align-items-center d-flex" >
+                        <i class="fa-solid fa-sun"></i>
+                        <div class="ms-2 form-check form-switch">
+                            <input class="form-check-input p-2" type="checkbox" role="switch" id="flexSwitchCheckChecked" checked onclick="switchTheme()"/>
```
Comment (kazet) [inline templates/components/navbar.jinja2:38]: the id is too generic, why not theme-switch ?

File: templates/components/navbar.jinja2 (hunk 2 above)
Comment (kshitijk4poor) [reply templates/components/navbar.jinja2:17]: yes, already tested

File: templates/components/navbar.jinja2 [hunk 4]
```diff
@@ -1,35 +1,43 @@
...
+    <nav class="navbar navbar-expand-lg bg-light" data-bs-theme="light">
+        <div class="container-fluid">
             <a class="navbar-brand" href="/"><img src="/static/images/logo.png" class="logo"></a>
-            <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation">
+            <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarColor01" aria-controls="navbarColor01" aria-expanded="false" aria-label="Toggle navigation">
                 <span class="navbar-toggler-icon"></span>
             </button>
-            <div class="collapse navbar-collapse">
-                <ul class="navbar-nav mr-auto">
+            <div class="collapse navbar-collapse" id="navbarColor01">
```
Comment (kazet) [inline templates/components/navbar.jinja2:8]: why the ID is navbarColor01 and not e.g. navbar?

File: templates/components/navbar.jinja2 [hunk 5]
```diff
@@ -1,35 +1,43 @@
...
+                        <a class="nav-link {{ 'active' if request.path == '/results' }}" href="/results?task_filter=interesting">View raw task results</a>
                     </li>
                     <li class="nav-item">
-                        <a class="nav-link" href="/export">Export reports</a>
+                        <a class="nav-link {{ 'active' if request.path == '/export' }}" href="/export">Export reports</a>
                     </li>
                     <li class="nav-item">
-                        <a class="nav-link" href="/exports">View exported reports</a>
+                        <a class="nav-link {{ 'active' if request.path == '/exports' }}" href="/exports">View exported reports</a>
+                        <span class="visually-hidden">(current)</span>
```
Comment (kazet) [inline templates/components/navbar.jinja2:24]: what this (current) does ? is this needed?

File: templates/components/navbar.jinja2 [hunk 6]
```diff
@@ -1,35 +1,42 @@
 <header>
-    <nav class="navbar navbar-expand-md navbar-light bg-light">
-        <div class="container">
+    <nav class="navbar navbar-expand-lg bg-light" data-bs-theme="light">
+        <div class="container-fluid">
             <a class="navbar-brand" href="/"><img src="/static/images/logo.png" class="logo"></a>
-            <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation">
+            <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarColor01" aria-controls="navbarColor01" aria-expanded="false" aria-label="Toggle navigation">
```
Comment (kazet) [inline templates/components/navbar.jinja2:5]: navbarcolor01 is still here ;)

**Review by kazet — APPROVED** [review]

**Review by kazet — APPROVED** [review]

**Pull request merged**